### PR TITLE
Split up "Doing science..." message

### LINKED ISSUE #39503: [Epic] Make “Doing Science” less ambiguous for long-running queries
- [product doc](https://www.notion.so/metabase/Make-Doing-Science-less-ambiguous-for-long-running-queries-b720f9d9ec144afcbda2edc463f2e596)
- feature branch: `39503-less-ambiguous-loading-state` 
- issues: #31135

**Implementation Plan**

```[tasklist]
### 1. Doing Science
- [ ] https://github.com/metabase/metabase/issues/39615
```

```[tasklist]
### 2. Dashboard cards
- [ ] https://github.com/metabase/metabase/issues/39620
```

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/39503
### Description

Allows the `getLoadingMessage` option to return a function which is parametrised on the elapsed query time.

### How to verify

1. Open a new query that loads very slowly
2. Verify that the loading message changes from `Doing Science...` to `Talking to the database...` after 4 seconds

### Demo
<div>
    <a href="https://www.loom.com/share/a05eea3db62c47b0b9e2ce010c8d33e6">
      <p>Question · Metabase - 6 March 2024 - Watch Video</p>
    </a>
</div>

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/table.cy.spec.js [hunk 1]
```diff
@@ -325,6 +325,28 @@ describe("scenarios > visualizations > table", () => {
       expect(isScrollableHorizontally($popover[0])).to.be.false;
     });
   });
+
+  it(
```
Comment (ranquild) [inline e2e/test/scenarios/visualizations-tabular/table.cy.spec.js:329]: there is `cy.clock` which can be used to advance time

File: frontend/src/metabase/query_builder/components/QueryVisualization.jsx [hunk 2]
```diff
@@ -98,22 +83,30 @@ export const VisualizationEmptyState = ({ className }) => (
...
-    </h2>
-  </div>
-);
+export function VisualizationRunningState({ className = "", loadingMessage }) {
+  const [isSlow] = useTimeout(SLOW_MESSAGE_TIMEOUT);
+
+  // show the slower loading message only when the loadingMessage is
+  // not customised
+  const message =
+    loadingMessage === "Doing science..." && isSlow()
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/QueryVisualization.jsx:92]: Few things:
- The message should be translated
- This logic is very easy to break

Wdyt if we do the following:
- Change `getLoadingMessage()` to `getLoadingMessage(isSlow: boolean)`
- Apply the changes in the default plugin
- Apply the changes in the EE plugin. If the setting value (enum value) is still `doing-science`, only then change the message
- Remove `loadingMessage: getWhiteLabeledLoadingMessage(state)` from the QueryBuilder component. Where needed, use the selector directly, e.g. with `useSelector`.

**Review by ranquild — CHANGES_REQUESTED** [review]
Comments inline

File: frontend/src/metabase/query_builder/components/QueryVisualization.jsx (hunk 2 above)
Comment (romeovs) [reply frontend/src/metabase/query_builder/components/QueryVisualization.jsx]: Ok, I've implemented you proposal, please have another look.

**Review by ranquild — APPROVED** [review]
LGTM but maybe after removing props drill it would be not hard to make it `getLoadingMessage(state, isSlow)` and not `(state)(isSlow)`. But not sure if it would be cleaner

**Pull request merged**